internal/metric.go

```diff
@@ -0,0 +1,15 @@
+package internal
```

Comment:
Creating an internal package is not adding any benefit. It is encapsulating already un-exported structures and does not add naming/functional benefit.